bmi_model.py

```diff
@@ -56,3 +60,78 @@ def update(self):
     def finalize(self):
         """Clean up any internal resources of the model"""
         pass
+
+    def get_component_name(self) -> str:
+        """Name of this BMI module component.
+
+        Returns:
+            str: Model Name
+        """
+        return "Tensor Test"
+
+    def get_input_item_count(self) -> int:
+        """Number of model input variables
+
+        Returns:
+            int: number of input variables
+        """
+        return len(self.input_names)
+    
+    def get_input_var_names(self) -> tuple[str]:
+        """The names of each input variables
+
+        Returns:
+            tuple[str]: iterable tuple of input variable names
+        """
+        return self.input_names
+
+    def get_output_item_count(self) -> int:
+        """Number of model output variables
+
+        Returns:
+            int: number of output variables
+        """
+        return len(self.output_names)
+    
+    def get_output_var_names(self) -> tuple[str]:
+        """The names of each output variable
+
+        Returns:
+            tuple[str]: iterable tuple of output variable names
+        """
+        return self.output_names
+
+    # BMI Variable Query
+    def get_value_ptr(self, name: str) -> ndarray:
```

Comment:
This is what I mean by "goes against BMI" to maintain a gradient. We don't have to return a `ndarray` and can return a `tensor` instead bc Python doesn't care about types if we want

Reply:
Using an `ndarray` is about the only way to ensure ABI compatibility if you want to use the python model's BMI from a non-python caller (e.g. ngen)